### PR TITLE
feat(docs): add guide generation from READMEs and guides in packages

### PR DESCRIPTION
**Pull request by damienwebdev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Currently, we don't generate documentation for easy consumption in daffio. 

Touches: #336 

## What is the new behavior?
This adds documentation generation for guides in packages.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-tree.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  constructor(key: string = '', title: string = '', path: string = '', children: NavigationTree[] = []) {
+    this.id = key;
+    this.title = title;
+    if(path){
+      this.path = path;
+    }
+    this.children = children;
+  }
+
+  static generateTreeFromDocuments(items: NavigationDocument[] = []) {
```
Comment (lderrickable) [inline tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-tree.ts]: I feel like this is a weird function for the `NavigationTree` to have. I'd rather it create a new tree upon construction, then maybe have an `insertMany` function.

File: tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-tree.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    for (let doc of items) {
+      tree.insert(doc.id, doc);
+    }
+    return tree;
+  }
+
+  /**
+   * Note that this is implemented recursively, so for a large number
+   * of nested documents, this will blow up. We don't anticipate 
+   * this happening, as the depth of the trie should be
```
Comment (lderrickable) [inline tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-tree.ts]: *tree

File: tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-tree.ts (hunk 1 above)
Comment (damienwebdev) [reply tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-tree.ts]: I agree. I felt this function was odd as a static member. I was considering it as part of the constructor, but it felt too complex. Do you think it makes sense as a separate pure function?

File: tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-tree.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+ */
+export class NavigationTree {
+  id: string = '';
+  path?: string;
+  title: string = '';
+  children: NavigationTree[] = [];
+
+  // Add the document to the `documents` of the relevant child
+  //Check to see if there's a parent document that needs to moved down,
+  //If there is, shift it down.
```
[the remark this answers is not in the document]
Comment (damienwebdev) [reply tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-tree.ts]: I think I should move it. It describes the special append algorithm that I use to decide exactly how to append a node, as there are a few special considerations.

File: tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-tree.ts (hunk 1 above)
Comment (damienwebdev) [reply tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-tree.ts]: I've moved this into a pure function.

File: tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-tree.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (damienwebdev) [reply tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-tree.ts]: I've moved the functions that describe the behavior into a functions within the class and documented them.

**Comment by lderrickable** [comment]
lgtm

**Pull request merged**